### ISSUE TITLE
Setting Not Retained After Exit on Fedora 33

### ISSUE THREAD

**Issue by bayazidbh** [eneshecan/whatsapp-for-linux#106]
**Description**
Cannot retain settings after exit or reboot on the snap version of Fedora 33. Desktop Notification, Close to Tray, and Start in Tray always started disabled after exit / when opened.

**To Reproduce**
1. Open the App (used menu item generated by snap with command "env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/whatsapp-for-linux_whatsapp-for-linux.desktop /var/lib/snapd/snap/bin/whatsapp-for-linux")
2. Turn on Desktop Notification, Close to Tray, and Start in Tray .
3. Exit / Shutdown
4. Re-open the App

**Expected behavior**
Settings retained between each run of the app.

**Screenshots**
![Screenshot from 2021-04-19 11-15-58](https://user-images.githubusercontent.com/26621899/115181703-af6b8d80-a102-11eb-828a-44ab9f433304.png)
![Screenshot from 2021-04-19 11-16-49](https://user-images.githubusercontent.com/26621899/115181709-b1355100-a102-11eb-8832-e03bcbde1ef4.png)

**Environment**
OS: Fedora release 33 (Thirty Three) x86_64 
Host: 80E4 Lenovo G40-80 
Kernel: 5.11.13-200.fc33.x86_64 
Uptime: 3 hours, 18 mins 
Packages: 2831 (rpm), 37 (flatpak), 9 (snap) 
Shell: bash 5.0.17 
Resolution: 1366x768 
DE: GNOME 3.38.4 
WM: Mutter 
WM Theme: WhiteSur-dark 
Theme: WhiteSur-dark [GTK2/3] 
Icons: BigSur-dark [GTK2/3] 
Terminal: gnome-terminal 
CPU: Intel i5-5200U (4) @ 2.700GHz 
GPU: AMD ATI Radeon HD 8550M / R5 M230 
GPU: Intel HD Graphics 5500 
Memory:  7865MiB 
snapd: 2.49-2.fc33

Snap Installed:
Name				| Version					| Rev	| Tracking	Publisher	Notes
authy				| 1.8.3					| 5      | latest/beta    | twilio-authy  | -
core18				| 20210309				| 1997   | latest/stable  | canonical✓    | base
core20				| 20210319 				| 975    | latest/stable  | canonical✓    | base
gnome-3-28-1804		| 3.28.0-19-g98f9e67.98f9e67	| 145    | latest/stable  | canonical✓    | -
gtk-common-themes	| 0.1-50-gf7627e4			| 1514   | latest/stable  | canonical✓    | -
snapd				| 2.49.2					| 11588  | latest/stable  | canonical✓    | snapd
whatsapp-for-linux		| 1.1.6					| 23     | latest/stable  | nshecan       | -
whitesur-gtk-theme	| 3.4						| 60     | latest/stable  | spider623     | -

**Comment by eneshecan** [comment]
I thought this is a problem with snap and couldn't reproduce it. Could you please check the log messages via starting the app from terminal?

**Comment by bayazidbh** [comment]
Ah, it seems to be some permission settings? I don't know why it doesn't automatically have the required permissions, however. I generally run it from the auto-generated menu item (hence the env string) though running `snap run whatsapp-for-linux` yields the same log messages.

```$ env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/whatsapp-for-linux_whatsapp-for-linux.desktop /var/lib/snapd/snap/bin/whatsapp-for-linux
WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement
Settings: Key file does not have key “allow_permissions” in group “Network”
Settings: Key file does not have key “close_to_tray” in group “General”
Settings: Key file does not have key “start_in_tray” in group “General”
Settings: Key file does not have key “header_bar” in group “General”
Settings: Key file does not have key “start_in_tray” in group “General”

** (WebKitWebProcess:13598): WARNING **: 08:35:17.832: WebKit wasn't able to find the GL video sink dependencies. Hardware-accelerated zero-copy video rendering can't be enabled without this plugin.
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
```

**Comment by eneshecan** [comment]
Could you please check if the setting file exists? Should be somewhere `./snap/whatsapp-for-linux/23/.config/whatsapp-for-linux/settings.conf`

**Comment by bayazidbh** [comment]
It exists, but the content is only as follow:

```
[General]
zoom_level=1

[Network]
```

**Comment by eneshecan** [comment]
Yes, they are missing somehow. Strange...

**Comment by antaxiom** [comment]
It seems it resets the config file every single time so even if I manually add the needed information I get this error.

**Comment by antaxiom** [comment]
If I add the needed variables into the config and chmod 0444 the file into a readonly state I cease getting errors, but cannot access past the QR screen. My understanding of this problem seems to stem from the "allow_permissions=true" variable not being recognized, therefore the application does not access the internet and I cannot use it.

**Comment by hakkikaancaliskan** [comment]
I have exact same issue on Arch. Installed via AUR.
If i add missing keys (close_to_tray, start_in_tray, header_bar, allow_permissions) to settings.conf, it works fine.
```
[General]
zoom_level=1
close_to_tray=true
start_in_tray=true
header_bar=true

[Network]
allow_permissions=true
```

AUR helper: yay
Operating System: Arch Linux
KDE Plasma Version: 5.22.4
KDE Frameworks Version: 5.84.0
Qt Version: 5.15.2
Kernel Version: 5.13.9-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 8 × Intel® Core™ i5-8250U CPU @ 1.60GHz
Memory: 11.6 GiB of RAM
Graphics Processor: Mesa Intel® UHD Graphics 620

**Comment by eneshecan** [comment]
Possibly fixed by aeaa278
